snapcraft/internal/pluginhandler/__init__.py

```diff
@@ -630,9 +628,6 @@ def _generate_part_env(self, step: steps.Step) -> str:
         else:
             plugin_environment = dict()
 
-        # Part's (user) say.
-        user_build_environment = self._part_properties["build-environment"]
```

Comment:
code suggestions not working for removed lines, I would keep user_build_environment but = self.build_environment

This makes the code obvious to casual readers. The "user" part of this variable name was your idea ;-)

Reply:
Makes sense, will do

Reply:
Done.